pkg/vmimage/data/validate.sh

```diff
@@ -21,19 +21,26 @@ EOF
 cp data/etc/yum.repos.d/kickstart.repo /etc/yum.repos.d/kickstart.repo
 
 yum check-update > /tmp/yum_check_update || true
-yum updateinfo > /tmp/yum_update_info || true
+yum updateinfo > /tmp/yum_updateinfo || true
+yum updateinfo list security -q > /tmp/yum_updateinfo_list_security || true
 
 
 # install openscap and run
 yum install -y openscap-scanner openscap-utils scap-security-guide python-lxml
 
 oscap xccdf eval \
   --profile xccdf_cloud.osadev_profile_stig_customized_aro \
-  --results /tmp/scap-results.xml \
-  --report /tmp/scap-report.html \
+  --results /tmp/scap_results.xml \
+  --report /tmp/scap_report.html \
   --tailoring-file /root/data/ssg-rhel7-ds-aro.xml \
   --oval-results --fetch-remote-resources \
   --cpe /usr/share/xml/scap/ssg/content/ssg-rhel7-cpe-dictionary.xml \
   /usr/share/xml/scap/ssg/content/ssg-rhel7-ds.xml > /tmp/oscap.log || true
   
-python -c 'import sys; from lxml import html; h = html.fromstring(open("/tmp/scap-report.html").read()); sys.exit(1) if len(h.find_class("rule-result rule-result-fail")) > 0 else sys.exit(0)'
+python -c 'import sys; from lxml import html; h = html.fromstring(open("/tmp/scap_report.html").read()); sys.exit(1) if len(h.find_class("rule-result rule-result-fail")) > 0 else sys.exit(0)'
+
+if [ -s "/tmp/yum_updateinfo_list_security" ]; then
+    >&2 echo "Found pending security updates:"
+    >&2 cat /tmp/yum_updateinfo_list_security
+    exit 1
```

Comment:
Hm, do we want this to fail CI? That might be overkill.

/cc @kwoodson

Reply:
In general, it works similarly to the scap report, but it shouldn't really fail when we run `make vmimage`, because we run `yum -y update` as part of the kickstart while building an image.

In case of `make vmimage-validate` on an older image, I would prefer it to fail as an indicator that some of the checks do not pass.

Why do you think we should not fail here? I feel like I'm missing something, so I would appreciate some input.

Reply:
We might need to move `python -c '...'` and this check to `vmimage.sh` and `vmimage-validate.sh`. So we do not perform any check in CSE script. Instead we will be grabbing logs from a VM and perform check on the higher level (in `vmimage.sh` and `vmimage-validate.sh`).

As a result we will have logs (scap report, npm opuputs, etc) in the build artifacts. It will make debugging easier.

/hold